### PR TITLE
Create university-of-pretoria-harvard-theology-religion.csl

### PR DESCRIPTION
**Pull request by schlittenhardtm**
A CSL according to the University of Pretoria Harvard referencing guide by the Faculty of Theology and Religion, does behave differently to typical Harvard regarding internet sources.

### CODE REVIEW

File: university-of-pretoria-harvard-theology-religion.csl [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+        </choose>
+      </substitute>
+    </names>
+  </macro>
+  <macro name="author-count">
+    <names variable="author">
+      <name form="count"/>
+    </names>
+  </macro>
+  <macro name="date-issued">
```
Comment (adam3smith) [inline university-of-pretoria-harvard-theology-religion.csl:93]: What's the goal here? This seems like an odd way to establish in press: page ranges can have roman numerals and the page field can include e-identifiers, neither of which would be "numeric"

File: university-of-pretoria-harvard-theology-religion.csl [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+      </if>
+      <else>
+        <text variable="title"/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="book-details">
+    <group delimiter=", ">
+      <group delimiter=" ">
+        <number variable="edition" form="ordinal"/>
```
Comment (adam3smith) [inline university-of-pretoria-harvard-theology-religion.csl:179]: worthwhile having coding for non-numeric editions in the style -- that's fairly common in user data (most existing style have an `if is-numeric="edition"` test)

File: university-of-pretoria-harvard-theology-religion.csl [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+    <author>
+      <name>Micha Schlittenhardt</name>
+      <email>micha.schlittenhardt@tuks.co.za</email>
+    </author>
+    <category citation-format="author-date"/>
+    <category field="generic-base"/>
+    <updated>2022-06-22T00:06:00+00:00</updated>
+    <rights license="http://creativecommons.org/licenses/by-sa/3.0/">This work is licensed under a Creative Commons Attribution-ShareAlike 3.0 License</rights>
+  </info>
+  <locale xmlns="http://purl.org/net/xbiblio/csl" xml:lang="en-GB">
```
Comment (adam3smith) [inline university-of-pretoria-harvard-theology-religion.csl:19]:
```suggestion
  <locale xml:lang="en-GB">
```

no need to restate the namespace

**Review by adam3smith — CHANGES_REQUESTED** [review]
Thanks! Looks great overall. A couple of small questions & comments

File: university-of-pretoria-harvard-theology-religion.csl (hunk 1 above)
Comment (schlittenhardtm) [reply university-of-pretoria-harvard-theology-religion.csl:93]: Thank you for your feedback, I am very new to CSL. How can I resolve this issue?

Comment (adam3smith) [reply university-of-pretoria-harvard-theology-religion.csl:93]: The standard way would be to test for date being empty and then use status, so along the lines of

```
<if variable="issued" match="none">
  <text variable="status"/>
</if>
```

In Zotero, e.g., Status would be entered as `status: in press` in the Extra field. 

That also allows for more flexibility, e.g., "forthcoming," "in press" etc.

File: university-of-pretoria-harvard-theology-religion.csl [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+      </if>
+      <else-if variable="page" match="none">
+        <text macro="date-issued-year"/>
+      </else-if>
+      <else-if is-numeric="page">
+        <text macro="date-issued-year"/>
+      </else-if>
+      <else-if variable="issued" match="none">
+        <text variable="status"/>
+      </else-if>
```
Comment (adam3smith) [inline university-of-pretoria-harvard-theology-religion.csl:106]:
```suggestion
      <else-if variable="issued" match="any">
        <text macro="date-issued-year"/>
      </else-if>
      <else>
        <text variable="status"/>
      </else>
```

How is this instead -- less complicated testing going on here. I'm particularly worried about the page="none" condition -- presumably, most in press articles don't, in fact, have page ranges

**Comment by adam3smith** [comment]
Thanks -- I have one quick suggestion above, rest looks good

**Review by schlittenhardtm — COMMENTED** [review]
I think all changes are applied now

**Comment by adam3smith** [comment]
Thanks -- looks great!

**Pull request merged**